VM/rel2vm_params.py

```diff
@@ -0,0 +1,1085 @@
+        qclogging.get_logger(logger_name), srf_meta["name"]
+    )
+
+    if not os.path.isdir(out_dir):
+        logger.debug(
+            "Output directory {} does not exist. Creating it now.".format(out_dir)
+        )
+        os.makedirs(out_dir)
+
+    # temp directory for current process
+    ptemp = mkdtemp(prefix="_tmp_%s_" % (srf_meta["name"]), dir=out_dir)
+
+    vm_params_dict = {}
+    vm_params_dict_extended = optimise_vm_params(
+        srf_meta, options, out_dir, ptemp, logger=logger
+    )
+    if vm_params_dict_extended is not None:
+        # store configs
+        vm_working_dir, nzvm_cfg, vm_params_path = temp_paths(ptemp)
+
+        # NZVM won't run if folder exists
+        # if os.path.exists(vm_working_dir):
+        #     logger.debug("VM working directory {} already exists.".format(vm_working_dir))
+        #     rmtree(vm_working_dir)
+
+        vm_params_dict = save_vm_params_yaml(
+            vm_params_path=vm_params_path,
+            vm_dir=vm_working_dir,
+            origin=vm_params_dict_extended["origin"],
+            rot=vm_params_dict_extended["bearing"],
+            xlen=vm_params_dict_extended["xlen_mod"],
+            ylen=vm_params_dict_extended["ylen_mod"],
+            zmax=vm_params_dict_extended["zlen_mod"],
+            hh=options["hh"],
+            min_vs=options["min_vs"],
+            mag=faultprop.Mw,
+            centroid_depth=srf_meta["hdepth"],
+            sim_duration=vm_params_dict_extended["sim_time_mod"],
+            topo_type=options["vm_topo"],
+            model_version=options["vm_version"],
+        )
+        if vm_params_dict:
+            # save important files
+            logger.debug("Creating directory {}".format(vm_working_dir))
+            os.makedirs(vm_working_dir)
+            # move(nzvm_cfg_path, vm_working_dir)
+            vm_params_path_final = os.path.join(
+                out_dir, os.path.basename(vm_params_path)
+            )
+            if os.path.exists(vm_params_path_final):
+                logger.info(
+                    f"Warning: {vm_params_path_final} already exists and to be overwritten."
+                )
+                os.remove(vm_params_path_final)
+
+            move("{}".format(vm_params_path), out_dir)
+            logger.info("Wrote vm_params.yaml at {}".format(out_dir))
+            if plot_enabled:
+                plot_vm(
+                    vm_params_dict_extended,
+                    srf_meta["corners"],
+                    faultprop.Mw,
+                    out_dir,
+                    ptemp,
+                    logger=logger,
+                )
+
+            print(f"Success: Wrote vm_params.yaml at {out_dir}", file=sys.stderr)
+            # generate a corners like NZVM would have
+            logger.debug("Saving VeloModCorners.txt")
+            with open("{}/VeloModCorners.txt".format(out_dir), "wb") as c:
+                c.write("> VM corners (python generated)\n".encode())
```

Comment:
As a note, I believe this format isn't consistent with the format that the binary file generates. It would be good to keep these consistent.

Reply:
What was the reason for this file to be generated in the first place?

Reply:
For the no-vm parameter. So you can see the corner output without having to generate the VM.

Reply:
Happy?
`>Velocity model corners(python generated)
>Lon	Lat
177.649867	-37.217773
174.830772	-37.179174
174.736805	-39.416560
177.644285	-39.456369`

Reply:
and we don't need to keep both versions? rel2vm_params will write one, then when vm_params2vm runs, the previous one will be overwritten

Reply:
Nope, shouldn't need to keep both. As long as scripts that run over the corners can operate on either output it'll be fine. 

If anything, we don't need the binary output but it's not worth changing that.